config/config.go

```diff
@@ -1,24 +1,34 @@
 package config
 
-import "time"
+import (
+	"time"
+
+	"github.com/joho/godotenv"
+	"github.com/kelseyhightower/envconfig"
+	"github.com/pkg/errors"
+)
 
 // Config stores all configurations of the application.
 // The values are read from a file or environment variables.
 type Config struct {
-	ServerAddress           string
-	ServerWriteTimeout      time.Duration
-	ServerReadTimeout       time.Duration
-	ServerIdleTimeout       time.Duration
-	GracefulShutdownTimeout time.Duration
+	ServerAddress           string        `envconfig:"SERVER_ADDRESS"`
+	ServerWriteTimeout      time.Duration `envconfig:"SERVER_WRITE_TIMEOUT"`
+	ServerReadTimeout       time.Duration `envconfig:"SERVER_READ_TIMEOUT"`
+	ServerIdleTimeout       time.Duration `envconfig:"SERVER_IDLE_TIMEOUT"`
+	GracefulShutdownTimeout time.Duration `envconfig:"GRACEFUL_SHUTDOWN_TIMEOUT"`
 }
 
-// Load reads configuration from file or environment variables.
-func Load(path, name string) (Config, error) {
-	return Config{
-		ServerAddress:           "0.0.0.0:8085",
-		ServerWriteTimeout:      15 * time.Second,
-		ServerReadTimeout:       15 * time.Second,
-		ServerIdleTimeout:       60 * time.Second,
-		GracefulShutdownTimeout: 30 * time.Second,
-	}, nil
+// Load returns configuration from file or environment variables.
+// Values from file take precedence over environment variables.
+func Load(path string) (*Config, error) {
```

Comment:
it's good we are reading config variables from config file. Don't you think we should also add support to read from environment variables too?

Reply:
![Uploading image.png…]()
already taken care of

Reply:
(y)